release notes/v0.38.0.md

```diff
@@ -0,0 +1,166 @@
+
+```bash
+# No color output
+K6_NO_COLOR=true k6 run script.js
+
+# No color output
+NO_COLOR= k6 run script.js
+```
+
+### Support for encrypted TLS private keys
+
+Now you can use passphrase-protected private keys when authenticating with TLS. Using the `password` property of an options' `tlsAuth` object, you can now indicate the passphrase to use to decrypt a private key. Note that this support is limited to the scope of [RFC1423](https://datatracker.ietf.org/doc/html/rfc1423) and does not support PKCS8 keys, as they're not yet supported by the golang standard library. 
+
+```javascript
+export const options = {
+  tlsAuth: [
+    {
+      domains: ['example.com'],
+      cert: open('mycert.pem'),
+      key: open('mycert-key.pem'),
+      password: 'mycert-passphrase',
+    },
+  ],
+};
+```
+
+Thanks, @Gabrielopesantos, for the [contribution](https://github.com/grafana/k6/pull/2488).
+
+### Improve JSON output's performance ([#2436](https://github.com/grafana/k6/pull/2436))
+
+The JSON output was optimized and now should around 2x as more performant at outputting metrics.  This means that it either will be able to export twice as many metrics and to use half the resources to do the same amount of metrics. As a side effect, there is a slight breaking change in which the `tags` field is no longer sorted, while it was previously.
+
+### Treat panics as interrupt errors ([#2453](https://github.com/grafana/k6/pull/2453))
+
+We changed the behavior of how k6 treats Go panics, which may happen because of bugs in k6 or in a JavaScript k6 extension. Previously, the behavior was to catch the panic and log it as an error. Starting with **v0.38.0**, whenever k6 observes a Go panic, it logs an error like before, but more importantly, it will abort the script execution and cause k6 to exit with a non-0 exit code. This will help extension authors to identify issues in their extensions more easily. 
+
+### Miscellaneous
+
+* [#2411](https://github.com/grafana/k6/pull/2411) The k6 command-line UI (logo, test description and progerss bars) can now effectively be disabled using the [`--quiet`](https://k6.io/docs/using-k6/options/#quiet) flag.
+* [#2429](https://github.com/grafana/k6/pull/2429) `lib/types` now exposes the source of the `NullHostnameTrie` to simplify access to an original list of the hostnames.
+* [#2484](https://github.com/grafana/k6/pull/2484) moved out in a new dedicated Go `lib/netext/grpcext` package all the parts not strictly required from the `js/k6/net/grpc` module for binding the gRPC operations and the JavaScript runtime. It facilitates the development of extensions based on gRPC without the direct dependency on the `goja` runtime. Furthermore, the new [Dial](https://github.com/grafana/k6/blob/bef458906f6884a99843573223028981c0a8b8db/lib/netext/grpcext/conn.go#L72-L81) function accepts a [grpc.DialOption](https://pkg.go.dev/google.golang.org/grpc@v1.45.0#DialOption) variadic for customizing the dialing operation.
```

Comment:
This should be moved to the extension section below, under a header like this:

### gRPC module refactored to enable gRPC extensions to use it